server/datastore/mysql/policies.go

```diff
@@ -465,6 +465,34 @@ func (ds *Datastore) TeamPolicy(ctx context.Context, teamID uint, policyID uint)
 	return policyDB(ctx, ds.reader(ctx), policyID, &teamID)
 }
 
+func (ds *Datastore) TeamNameByPolicyName(ctx context.Context, policyName string) (rowFound bool, isGlobal bool, teamName string, err error) {
```

Comment:
Currently, we have methods like `QueryByName`, `TeamByName`, `LabelIDsByName`, etc.

How about defining a `PolicyByName(ctx context.Context, name string) (*Policy, error)` (For reusability.)
- `rowFound` could be determined if `err != nil && fleet.IsNotFound(err)`
- `isGlobal` could be determined if `policy.TeamID == nil`
- `teamName` can be obtained with `ds.Team(id)`

Reply:
thx, I was over-thinking avoiding the 2nd query and extra joins.  Updated.